containerfixture/containerfixture.go

```diff
@@ -0,0 +1,119 @@
+// many times as needed with dynamic values
+func modifyYaml(yaml string) map[string]string {
+	podName := fmt.Sprintf("test-pod-%s", GetRandomPort())
+	webPort := GetRandomPort()
+	apiPort := GetRandomPort()
+
+	yaml = strings.Replace(yaml, "name: \"test-pod\"", fmt.Sprintf("name: %s", podName), 1)
+	yaml = strings.Replace(yaml, "hostPort: 8000", fmt.Sprintf("hostPort: %s", webPort), 1)
+	yaml = strings.Replace(yaml, "hostPort: 8089", fmt.Sprintf("hostPort: %s", apiPort), 1)
+
+	return map[string]string{
+		"yaml":    yaml,
+		"podName": podName,
+		"webPort": webPort,
+		"apiPort": apiPort,
+	}
+}
+
+// Building and Running a k8s manifest using 'podman kube play' command.
+// It will make up to three attempts to run with different configurations,
+// allowing it to run concurrently if necessary
+func BuildAndRunPod() map[string]string {
+	var deployment map[string]string
+	retries := 3
+	for retry := 0; retry < retries; retry++ {
+		deployment := modifyYaml(manifest)
+
+		cmd := exec.Command("podman", "play", "kube", "-")
+		cmd.Stdin = strings.NewReader(deployment["yaml"])
+		output, err := cmd.CombinedOutput()
+		if err == nil {
+			outputString := strings.Split(strings.TrimSpace(string(output)), "\n")
+			containerID := outputString[len(outputString)-1]
+			deployment["containerID"] = containerID
+			return deployment
+		}
+		log.Printf("Failed to start pod (attempt %d of %d).\n", retry+1, retries)
+	}
+	Cleanup("", deployment["podName"])
+	log.Fatalf("Failed to start pod after %d retries", retries)
+	return make(map[string]string) // Avoid 'missing return' error
+}
+
+// VerifySplunkServiceIsUp continuously monitors the Splunk container to
+// ensure it is up and ready for use in tests. It achieves this by repeatedly
+// checking the status endpoint of the Splunk API over a period of two minutes.
+func VerifySplunkServiceIsUp(port string) {
+	timeoutStart := time.Now().Unix()
+	splunkURL := fmt.Sprintf("https://localhost:%s/services/server/status", port)
+
+	for {
+		cmd := exec.Command("curl", "-s", "-o", "/dev/null", "-w", "%{http_code}", "-u",
+			fmt.Sprintf("%s:%s", Username, Password), "--insecure", splunkURL)
+		output, _ := cmd.CombinedOutput()
+		if string(output) == "200" {
+			log.Println("Splunk service deployed successfully!")
+			break
+		}
+		if time.Now().Unix()-timeoutStart > 120 {
+			log.Fatalf("The Splunk instance is not up, cannot verify indexing for tests.")
+		}
+		time.Sleep(5 * time.Second)
+	}
+}
+
+// Making sure to stop and remove the container we deployed
+func Cleanup(containerID string, podName string) {
+	if containerID != "" {
+		log.Println("Stopping and removing the container...")
+		cmd := exec.Command("podman", "stop", containerID)
+		if err := cmd.Run(); err != nil {
+			log.Println("Error stopping the container:", err)
+		}
+		cmd = exec.Command("podman", "rm", "-f", containerID)
+		if err := cmd.Run(); err != nil {
+			log.Println("Error removing the container:", err)
+		}
+	}
```

Comment:
Why do we need this if `podName` is known and included in the return value of `BuildAndRunPod`?

Reply:
Only to shut it gracefully, I wanted to avoid removing them forcefully with `podman pod rm -f`. We can remove that, not sure if we shuld.